integration-cli/check_test.go

```diff
@@ -83,7 +83,240 @@ func ensureTestEnvSetup(t *testing.T) {
 
 func TestDockerSuite(t *testing.T) {
 	ensureTestEnvSetup(t)
-	suite.Run(t, &DockerSuite{})
+	suite.Run(t, &DockerAPISuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerBenchmarkSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliAttachSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliBuildSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliCommitSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliCpSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliCreateSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliEventSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliExecSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliHealthSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliHistorySuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliImagesSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliImportSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliInfoSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliInspectSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliLinksSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliLoginSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliLogsSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliNetmodeSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliNetworkSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPluginLogDriverSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPluginsSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPortSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliProxySuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPruneSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPsSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPullSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliPushSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliRestartSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliRmiSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliRunSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliSaveLoadSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliSearchSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliSniSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliStartSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliStatsSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliTopSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliUpdateSuite{ds: &DockerSuite{}})
+	suite.Run(t, &DockerCliVolumeSuite{ds: &DockerSuite{}})
+}
```

Comment:
> Nice! I'm still a little worried about all those duplicated tiny "test suite" function entrypoints

If it's about this function, it will be removed in a follow-up when linux tests will be moved to GHA so we can also use the `gotestlist` logic. I created this function temporarily just for backward compat with what Jenkins currently consumes: https://github.com/moby/moby/blob/20d6b5c1bdedffe74f9dd774e3e22ad59360a51c/Jenkinsfile#L328. As I don't have write access to this repo, making changes to the Jenkinsfile will not be reflected unfortunately but I could replace the `DockerSuite` with this list.